AnthropicEvaluator.py

```diff
@@ -0,0 +1,68 @@
+import os
+import tiktoken
+from LLMNeedleHaystackTester import LLMNeedleHaystackTester
+from anthropic import AsyncAnthropic, Anthropic
+
+
+class AnthropicEvaluator(LLMNeedleHaystackTester):
+    def __init__(self,**kwargs):
+        if 'anthropic_api_key' not in  kwargs and not os.getenv('ANTHROPIC_API_KEY'):
+            raise ValueError("Either anthropic_api_key must be supplied with init, or ANTHROPIC_API_KEY must be in env")
+
+        if 'model_name' not in kwargs:
+            raise ValueError("model_name must be supplied with init")
+        elif "claude" not in kwargs['model_name']:
+            raise ValueError(
+                "If the model provider is 'Anthropic', the model name must include 'claude'. "
+                "See https://docs.anthropic.com/claude/reference/selecting-a-model for more details on Anthropic models")
+
+        if 'evaluation_method' not in kwargs:
+            print("since evaluation method is not specified , default method substring_match will be used for evaluation")
+        elif kwargs['evaluation_method'] not in ('gpt4', 'substring_match'):
+            raise ValueError("evaluation_method must be 'substring_match' or 'gpt4'")
+        elif kwargs['evaluation_method'] == 'gpt4' and 'openai_api_key' not in  kwargs and not os.getenv('OPENAI_API_KEY'):
+            raise ValueError("if evaluation_method is gpt4 , openai_api_key must be supplied with init, or OPENAI_API_KEY must be in env")
+        else:
+            self.openai_api_key= kwargs.get('openai_api_key', os.getenv('OPENAI_API_KEY'))
+
+        self.anthropic_api_key = kwargs.pop('anthropic_api_key', os.getenv('ANTHROPIC_API_KEY'))
+        self.model_name = kwargs['model_name']
+        self.model_to_test_description = kwargs.pop('model_name')
+        self.model_to_test = AsyncAnthropic(api_key=self.anthropic_api_key)
+        self.tokenizer = Anthropic().get_tokenizer()
+
+
+        super().__init__(**kwargs)
+
+    def get_encoding(self,context):
+        return self.tokenizer.encode(context).ids
+
+    def get_decoding(self, encoded_context):
+        return self.tokenizer.decode(encoded_context)
+
+    def get_prompt(self, context):
+        with open('Anthropic_prompt.txt', 'r') as file:
```

Comment:
I noticed `OpenAIEvaluator` has [values](https://github.com/gkamradt/LLMTest_NeedleInAHaystack/pull/7/files#diff-11ecb358dc120bc9366ed9910773344e8249df3f4b3f45f3c0c1f279d538ce1bR51) like this hardcoded. To maintain consistency you may consider also having those values in `AnthropicEvaluator` and removing `Anthropic_prompt.txt`

Reply:
Good point, I have updated the code to add the prompt directly under `get_prompt` method. 

[Text Completion API](https://docs.anthropic.com/claude/reference/complete_post)  in Anthropic is deprecated, their recommendation is to use Message API. I have updated the code to use Message API following this [migration guide](https://docs.anthropic.com/claude/reference/migrating-from-text-completions-to-messages)